### PR TITLE
feat: establish the component mechanism of the iot system 

### LINKED ISSUE #1611: [feature request] Establish the component mechanism of the iot system
**What would you like to be added**:
Build the component mechanism of the iot system to support the expansion of components and improve the user experience.

**Why is this needed**:
Now we have support for advanced users to customize their iot systems, refer #1596. In [[Feature Request]provide minimal set of components by default to reduce resource consumption](https://github.com/orgs/openyurtio/projects/14?pane=issue&itemId=23562027), @wawlian gives the most concise set of edgex required components, we will be based on this set of solutions for iot system deployment by default, all other optional components will be included in the component mechanism.
For some novice users, we need to give them a more user-friendly way to deploy edgex extensions, for example:
```yaml
apiVersion: iot.openyurt.io/v1alpha2
kind: PlatformAdmin
metadata:
  name: edgex-sample
spec:
  version: minnesota
  poolName: beijing
  component:
  - edgex-ui
  - edgex-device-onvif-camera
  - edgex-app-metrics-influxdb
  ...
```
My current vision is to first build this mechanism on top of an enhanced auto-collector, bringing all of edgex-compose's officially supported extensions into openyurt. Secondly, we can also build tools like user-developed extensions, or incorporate extensions from other communities, which is the potential of this mechanism.
At the same time, how to balance user-defined configuration and component mechanism is also a problem, at present I think it is more reasonable to support extended components to provide users with modifications, but this aspect may bring some complex processing logic to the iot (especially when removing component), we can discuss these in the future.
**others**
/kind feature

### PR DESCRIPTION
**Pull request by LavenderQAQ**
#### What type of PR is this?
/kind feature
/sig iot

#### What this PR does / why we need it:
Build PlatformAdmin's component mechanism to allow users to deploy common edgex extensions by simply filling in the component name.

#### Which issue(s) this PR fixes:

Fixes #1611

### CODE REVIEW

**Comment by LavenderQAQ** [comment]
This pr is done, but it needs to wait for #1647 to merge.

**Comment by LavenderQAQ** [comment]
@wangxye @Rui-Gan Please review it for me.

**Comment by LavenderQAQ** [comment]
This pr is ready.

File: pkg/controller/platformadmin/config/EdgeXConfig/manifest.yaml [hunk 1]
```diff
@@ -2,9 +2,39 @@ updated: false
...
+  requiredComponents: 
+  - edgex-core-command
+  - edgex-core-consul
+  - edgex-core-metadata
+  - edgex-redis
+- name: minnesota
+  requiredComponents: 
+  - edgex-core-command
+  - edgex-core-consul
+  - edgex-core-metadata
```
Comment (Rui-Gan) [inline pkg/controller/platformadmin/config/EdgeXConfig/manifest.yaml:27]: Do we need to add the core-common-config-bootstrapper service?

Comment (Rui-Gan) [reply pkg/controller/platformadmin/config/EdgeXConfig/manifest.yaml:27]: After discussion, this component needs to be added.

**Review by Rui-Gan — CHANGES_REQUESTED** [review]

**Review by Rui-Gan — APPROVED** [review]

**Review by rambohe-ch — APPROVED** [review]

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**